stdlib/public/core/Collection.swift

```diff
@@ -398,6 +392,15 @@ extension Collection {
   }
 }
 
+/// Supply a default `next()` method for `Collection` models that
+/// use some model of `Strideable` as their `Index`.
+extension Collection where Index : Strideable {
+  @warn_unused_result
+  public func next(i: Index) -> Index {
+    return i.advanced(by: 1)
```

Comment:
Yes, we would like to have that.  I suspect that we would have to remove some of them for practical performance reasons, but let's add them to start.

Please use either:

```
// FIXME: swift-3-indexing-model: range check.
```

or `_failEarlyRangeCheck(_:bounds:)`